### ISSUE TITLE
[$1000] Web - In the IOU report room, DM message do not show up immediately after paying elsewhere

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#25834]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login as User A and User B in front of you at the same time.
2. As User A request Money from User B ( make sure it works by avoiding the bugs around IOU feature Now  )
3.  As User A click on the request money message to open the IOU report room.
4. Do the same as User B ( Open the IOU report )
5. As User B Click Pay elsewhere to pay the request.
6. Now return to User A and  start sending messages
7. Notice User B IOU report room

## Expected Result:
In the IOU report room, the message should appear immediately for user B when user A sends it.

## Actual Result:
In the IOU report room, the message does not show up immediately for User B after paying elsewhere. It will show up if you leave the room and return.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.3.57-0

**Reproducible in staging?:** Y

**Reproducible in production?:** Y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/ec4145b5-b46c-462f-b677-53b34ee0366b

https://github.com/Expensify/App/assets/115492554/3cb1b174-a5fa-46c3-b0c6-789317215aa9

**Expensify/Expensify Issue URL:**

**Issue reported by:** @Ahmed-Abdella

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692036000505899

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d390fe5a34f8e2fd</li>
        <li>Upwork Job ID: 1694875039292411904</li>
        <li>Last Price Increase: 2023-08-25</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @garrettmknight</details>

**Comment by michaelhaxhiu** [comment]
I think this bug report is valid, and am not sure if it's a distinct bug in itself or it belongs to a larger performance issue. My hunch is this is performance / regression related, because IIRC in the past these requests & messages were instant.

**Comment by michaelhaxhiu** [comment]
Let's get opinion from a C+

**Comment by windragon0910** [comment]
I think it is because of ayncronation of the request.My hunch is to solve this issue rebuilding the functionality of the statement.And checking the workflow.

**Comment by s-alves10** [comment]
This looks like a backend issue. After user B paid elsewhere, user B doesn't receive pusher events

**Comment by michaelhaxhiu** [comment]
Thanks all! Let's go `internal` and then get an engineer from the team to 👀  at it.

**Comment by michaelhaxhiu** [comment]
@techievivek would love your insight here. I commented [above](https://github.com/Expensify/App/issues/25834#issuecomment-1692600753):
>I think this bug report is valid, and am not sure if it's a distinct bug in itself or it belongs to a larger performance issue. My hunch is this is performance / regression related, because IIRC in the past these requests & messages were instant.

**Comment by techievivek** [comment]
It seems like there's a performance bottleneck causing delays in real-time updates. However, it may not be worth investigating as a standalone problem since it may be part of a larger issue. I suggest bringing it up for discussion and deciding on next steps. 

We have another performance related issue here https://github.com/Expensify/App/issues/24486 that can be investigated together? CC @alexpensify

**Comment by michaelhaxhiu** [comment]
Asking on that GH if the volunteer from Callstack thinks these can be grouped. Otherwise, I will push this forward separately (or start a discussion in slack like @techievivek suggested).

**Comment by michaelhaxhiu** [comment]
ah wait this is `internal` it seems. @techievivek would you want to coordinate with @OlimpiaZurek to resolve this and https://github.com/Expensify/App/issues/24486 together?

**Comment by OlimpiaZurek** [comment]
Just wanted to give you a heads-up - we are gathering all performance related issues at the moment and we are doing some audits. Therefore, we are waiting for this issue to be resolved until we agree on further steps to improve the app's performance.

**Comment by michaelhaxhiu** [comment]
@OlimpiaZurek sounds good! Let's update by EOW or early new week if possible.

**Comment by techievivek** [comment]
Hi, @OlimpiaZurek, do you have any updates here?

**Comment by OlimpiaZurek** [comment]
@techievivek I think both issues are related to multiple APIs and the unnecessary re-renders mentioned here: https://expensify.slack.com/archives/C049HHMV9SM/p1692661200803469

Here is a tracking issue for multiple PR’s and improvements which are found: 
https://github.com/Expensify/App/issues/26087

So I think we should put these issues on hold and retested them again once all the above performance issues are resolved.

**Comment by michaelhaxhiu** [comment]
@techievivek I'll await your review/input (on @OlimpiaZurek's investigation, above). If we agree then we'll need to:
1. Add `[Hold on #26087]` in the GH title
2. Swap `Daily` for `Weekly`

And then we'll stay tuned and check in on this for potential resolution / unblocking at that time. Just laying out these next steps so it's clearly defining the path forward based on the latest investigation.

**Comment by techievivek** [comment]
Yeah, I agree with the assessment. I think the first thing that we would want to optimize is multiple API calls and several re-renders, once we are done with that we can move to the backend optimization(which we are already working on).

**Comment by techievivek** [comment]
ON hold.

**Comment by michaelhaxhiu** [comment]
I'm re-assigning this to another BZ as part of my preparation for Sabbatical (starting Friday).

**Next steps:**

- This is on hold till https://github.com/Expensify/App/issues/26087 is finalized and closed ⏸️ 
- Lots of progress on https://github.com/Expensify/App/issues/26087 lately which is good, so keep an eye 
- Labelling this `Weekly` so you can keep a pulse on the blocker and update the title of this GH when it's ready for Vivek to push

**Comment by techievivek** [comment]
Still on HODL because above issues are not done yet: https://github.com/Expensify/App/issues/25834#issuecomment-1739213452

**Comment by tjferriss** [comment]
Still on hold. I'm going to pass this over to another BZ team member as I'm leaving the team.

**Comment by techievivek** [comment]
Still on HOLD, the linked issue is still in progress https://github.com/Expensify/App/issues/26087

**Comment by techievivek** [comment]
Still on HOLD ^

**Comment by techievivek** [comment]
Still on HOLD ^

**Comment by garrettmknight** [comment]
@techievivek it looks like that issue is still open, but only because it's awaiting payment? Just in case, I retested this one and the behavior's still occurring:
<img width="1312" alt="Screenshot 2023-10-18 at 2 20 35 PM" src="https://github.com/Expensify/App/assets/1529980/689ac526-38f7-47f0-b951-322b5b3d1ad8">

**Comment by garrettmknight** [comment]
@techievivek can you take a look at this one when you get a chance since it's still happening?

**Comment by techievivek** [comment]
Triaging it now.

**Comment by techievivek** [comment]
I am unable to reproduce this, it looks like this has been fixed internally? I will request the QA to have a look.


https://github.com/Expensify/App/assets/35863227/41ffab8c-ff19-4093-9004-3d7fe3c6d4e3

**Comment by techievivek** [comment]
Requested a re-testing here: https://expensify.slack.com/archives/C9YU7BX5M/p1698662596321289

**Comment by Ahmed-Abdella** [comment]
@techievivek It is still reproducible. The messages must be sent from the user who requested the money and they do not appear for the user who paid. In your video, you have implemented this in the opposite way.

**Comment by techievivek** [comment]
@Ahmed-Abdella I have tried sending messages as both payer and payee, and it seems that both parties are receiving the messages. Is there anything else I should be doing?

**Comment by Ahmed-Abdella** [comment]
@techievivek Maybe you send the messages as payee after sending some as payer. You should send the messages as a payee directely after the payer paid. 


https://github.com/Expensify/App/assets/77763699/0497121e-64a1-4b01-9e50-8bb71d220e57

Sorry for the quality of the video will upload another one if needed

**Comment by kbecciv** [comment]
Issue is not fix yet, build 1.3.93.0

https://github.com/Expensify/App/assets/93399543/1b458143-e18c-465c-8671-be020ce6af92

**Comment by techievivek** [comment]
Not overdue. It looks like the issue is stilll reproducible, I will take a look at out backend code to find out a fix for this. Thanks for testing it out @Ahmed-Abdella @kbecciv

**Comment by techievivek** [comment]
Not overdue, will take a look at it today.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by Ahmed-Abdella** [comment]
@garrettmknight @techievivek , It seems that the issue has been fixed. Could you please assist me with the reporting bonus when you have time?

**Comment by Ahmed-Abdella** [comment]
@garrettmknight @techievivek It seems that the issue has been fixed. Could you please assist me with the reporting bonus when you have time?

**Comment by garrettmknight** [comment]
@Ahmed-Abdella just sent you an offer in Upwork

**Comment by garrettmknight** [comment]
Updated

**Comment by Ahmed-Abdella** [comment]
Sorry @garrettmknight, It is still the same (50$)

**Comment by garrettmknight** [comment]
Updated

**Comment by Ahmed-Abdella** [comment]
I accepted the offer, Thanks @garrettmknight

**Comment by garrettmknight** [comment]
Paid, closing.
